### PR TITLE
google: Create only one network per namespace

### PR DESCRIPTION
**Pull request by kklin**
This PR makes it so we only create one network per Google namespace, and also does some small cleanups to the Google provider.

### CODE REVIEW

**Comment by ejj** [comment]
ooooh excited for this!

File: cluster/google/google.go [hunk 1]
```diff
@@ -565,7 +605,7 @@ func (clst *Cluster) createNetwork() error {
 
 	log.Debug("Creating network")
 	op, err := clst.gce.InsertNetwork(&compute.Network{
-		Name:      clst.ns,
+		Name:      clst.networkName,
```
Comment (luise) [inline cluster/google/google.go]: Shouldn't the argument to `networkExists` (line 596)  also change to `clst.networkName`?

File: cluster/google/google.go [hunk 2]
```diff
@@ -295,17 +295,50 @@ func (clst *Cluster) instanceNew(name string, size string,
...
+		fwApplies := false
+		for _, tag := range fw.TargetTags {
+			if tag == clst.zone {
+				fwApplies = true
+				break
+			}
+		}
+
+		if fwApplies {
+			fws = append(fws, *fw)
```
Comment (kklin) [inline cluster/google/google.go]: No reason, just didn't think of it! Good catch!

File: cluster/google/google.go (hunk 1 above)
Comment (kklin) [inline cluster/google/google.go]: Good catch, thanks! (The reason why it worked is that we happen to make the network name the namespace)

**Pull request merged**